sal/middleware-sal/src/main/java/org/openecard/mdlw/sal/MiddlewareSAL.java

```diff
@@ -742,6 +747,12 @@ public SignResponse sign(Sign request) {
 		    }
 
 		    byte[] sig = key.sign(message);
+
+		    // encode EC signatures as ASN.1 structure
+		    if (marker.getAlgorithmInfo().getAlgorithm().contains("withECDSA")) {
```

Comment:
While this is correct, we should use code which is typesafe. Meaning if the Algorithm IDs change (which has been the case before) we want everything to work without changes here.
Look at the getPKCS11Alg funtion used a few lines above. This should be changed to return a SignatureAlgorithms type. This can then be used to get the key type.